File: clients/java/src/main/java/model/ServerMessageGame.java
package model;

import util.StreamUtil;

public class ServerMessageGame {
    private model.PlayerView playerView;
    public model.PlayerView getPlayerView() { return playerView; }
    public void setPlayerView(model.PlayerView playerView) { this.playerView = playerView; }
    public ServerMessageGame() {}
    public ServerMessageGame(model.PlayerView playerView) {
        this.playerView = playerView;
    }
    public static ServerMessageGame readFrom(java.io.InputStream stream) throws java.io.IOException {
        ServerMessageGame result = new ServerMessageGame();
        if (StreamUtil.readBoolean(stream)) {
            result.playerView = model.PlayerView.readFrom(stream);
        } else {
            result.playerView = null;
        }
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        if (playerView == null) {
            StreamUtil.writeBoolean(stream, false);
        } else {
            StreamUtil.writeBoolean(stream, true);
            playerView.writeTo(stream);
        }
    }
}

File: clients/kotlin/src/main/kotlin/model/ServerMessageGame.kt
package model

import util.StreamUtil

class ServerMessageGame {
    var playerView: model.PlayerView? = null
    constructor() {}
    constructor(playerView: model.PlayerView?) {
        this.playerView = playerView
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): ServerMessageGame {
            val result = ServerMessageGame()
            if (StreamUtil.readBoolean(stream)) {
                result.playerView = model.PlayerView.readFrom(stream)
            } else {
                result.playerView = null
            }
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        val playerView = playerView;
        if (playerView == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            playerView.writeTo(stream)
        }
    }
}
